From 838b56a33a3b25a1e1816e38ba668bca635eeda7 Mon Sep 17 00:00:00 2001
From: paledega <38860643+paledega@users.noreply.github.com>
Date: Sat, 7 Sep 2019 08:40:39 +0300
Subject: [PATCH] Javac strip added

---
 core/java.mk | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/core/java.mk b/core/java.mk
index 914784948e..c81962d5a5 100644
--- a/core/java.mk
+++ b/core/java.mk
@@ -233,11 +233,8 @@ endif
 # For user / userdebug builds, strip the local variable table and the local variable
 # type table. This has no bearing on stack traces, but will leave less information
 # available via JDWP.
-ifneq (,$(PRODUCT_MINIMIZE_JAVA_DEBUG_INFO))
-ifneq (,$(filter userdebug user,$(TARGET_BUILD_VARIANT)))
-LOCAL_JAVACFLAGS+= -g:source,lines
-endif
-endif
+
+LOCAL_JAVACFLAGS+= -g:none
 
 # List of dependencies for anything that needs all java sources in place
 java_sources_deps := \
